src/pip/_internal/cli/main_parser.py

```diff
@@ -45,6 +49,47 @@ def create_main_parser() -> ConfigOptionParser:
     return parser
 
 
+def identify_python_interpreter(python: str) -> Optional[str]:
+    if python == "python" or python == "py":
+        # Run the active Python.
+        # We have to be very careful here, because:
+        #
+        # 1. On Unix, "python" is probably correct but there is a "py" launcher.
+        # 2. On Windows, "py" is the best option if it's present.
+        # 3. On Windows without "py", "python" might work, but it might also
+        #    be the shim that launches the Windows store to allow you to install
+        #    Python.
+        #
+        # We go with getting py on Windows, and if it's not present or we're
+        # on Unix, get python. We don't worry about the launcher on Unix or
+        # the installer stub on Windows.
+        py = None
+        if WINDOWS:
+            py = shutil.which("py")
+        if py is None:
+            py = shutil.which("python")
```

Comment:
On many linux boxes this will likely return python2.

Also, as a user trying the command blindly, I was surprised that `pip --python python` did work, but `pip --python python3` or `pip --python python3.9` did not.

Not sure what's best here. Perhaps we need an entirely different logic on Windows and Posix.

Reply:
I'm also not sure. This is frankly a pretty obscure corner case - people shouldn't *want* to point pip at the system Python, and if it's an activated virtual environment, the most likely cases are that pip's installed there, or you're using the zipapp, so `--python` is not needed. So maybe the right answer is to remove this case altogether. People can still supply the full path to the interpreter if they want.

Reply:
Removing this case feels like a reasonable move. This is only ever useful if the current shell has an active virtual environment, and in that case we should probably check for `os.environ["VIRTUAL_ENV"]` instead.

Reply:
Yep, I've bitten the bullet and removed this.